snap/patches/nova/0001-remotefs-http-driver.patch

```diff
@@ -0,0 +1,226 @@
++    cfg.StrOpt('file_server_client_cert_file',
++               help='Path to client certificate file for mutual TLS to the '
++                    'file server.'),
++    cfg.StrOpt('file_server_client_key_file',
++               help='Path to client private key file for mutual TLS to the '
++                     'file server.'),
+ ]
+ 
+ libvirt_volume_vzstorage_opts = [
+diff --git a/nova/virt/libvirt/volume/remotefs.py b/nova/virt/libvirt/volume/remotefs.py
+index 8ddcaa1fcc..8096c77903 100644
+--- a/nova/virt/libvirt/volume/remotefs.py
++++ b/nova/virt/libvirt/volume/remotefs.py
+@@ -24,6 +24,8 @@ from oslo_utils import importutils
+ import nova.conf
+ import nova.privsep.fs
+ from nova import utils
++from nova.virt.libvirt.volume.remotefs_http import HttpDriver as _HttpDriverImpl
++HttpDriver = _HttpDriverImpl
+ 
+ LOG = logging.getLogger(__name__)
+ 
+diff --git a/nova/virt/libvirt/volume/remotefs_http.py b/nova/virt/libvirt/volume/remotefs_http.py
+new file mode 100644
+index 0000000000..7e3ddb5abe
+--- /dev/null
++++ b/nova/virt/libvirt/volume/remotefs_http.py
+@@ -0,0 +1,157 @@
++import os
++import ssl
++import tempfile
++from typing import Tuple
++
++from oslo_log import log as logging
++
++import nova.conf
++from urllib.parse import urlsplit
++from webdav4.client import ResourceAlreadyExists, ForbiddenOperation
++
++
++LOG = logging.getLogger(__name__)
++CONF = nova.conf.CONF
++
++LOG.debug("[webdav] remotefs_http module loaded")
++
++
++def _parse_remote_dst(dst: str) -> Tuple[str, str]:
++    """Split a remote destination of the form \"<host>:<absolute_path>\"."""
++    if ":" not in dst:
++        raise ValueError(
++            f"Invalid remote path format, expected 'host:/abs/path', got: {dst}"
++        )
++    parsed = urlsplit(f"scheme://{dst}")
++    host = parsed.hostname
++    path = parsed.path
++
++    if not host or not path:
++        raise ValueError(
++            f"Invalid remote path format, expected 'host:/abs/path', got: {dst}"
++        )
++    if not path.startswith("/"):
++        raise ValueError(f"Destination path must be absolute, got: {path}")
++    return host, path
++
++
++def _ensure_trailing_slash(path: str) -> str:
++    if not path or path.endswith("/"):
++        return path
++    return f"{path}/"
++
++
++def _base_url(host: str) -> str:
++    port = getattr(CONF.libvirt, "file_server_port", 10099)
++    return f"https://{host}:{port}"
++
++
++def _new_client(host: str):
++    # Lazy import to keep dependency surface localized
++    from webdav4.client import Client
++
++    base_url = _base_url(host)
++
++    # TLS verification and client certs for mTLS
++    ca_file = getattr(CONF.libvirt, "file_server_ca_file", "") or True
++    cert_file = getattr(CONF.libvirt, "file_server_client_cert_file", "") or None
++    key_file = getattr(CONF.libvirt, "file_server_client_key_file", "") or None
++    client_cert = (cert_file, key_file) if cert_file and key_file else None
```

Comment:
nit: can avoid defining client_cert and directly use `if cert_file and key_file` in L#125